misk-crypto/src/main/kotlin/misk/crypto/CryptoModule.kt

```diff
@@ -96,7 +96,7 @@ class CryptoModule(
       newMultibinder<ExternalKeySource>()
       multibind<ExternalKeySource>().to<S3KeySource>()
 
-      keyManagerBinder.addBinding().to<ExternalKeyResolver>()
+      keyManagerBinder.addBinding().to<ExternalKeyResolver>().asEagerSingleton()
```

Comment:
Better use `.in(Singleton::class)`.
Using `.asEagerSingleton()` means the class will be instantiated immediately, which will also force all object it depends on to be instantiated as well.